File: Project/iOS/User/eduku/Models/RecentUpdates.swift
//
//  RecentUpdates.swift
//  eduku
//
//  Created by Mickey on 2/17/16.
//  Copyright © 2016 Mickey. All rights reserved.
//

import Foundation
import Parse

class RecentUpdates {
    var recentLog: NSMutableArray
    
    var recentUpdates: PFObject!
    
    init(){
        recentLog = NSMutableArray()
    }
    init(_recentUpdates: PFObject) {
        recentUpdates = _recentUpdates
        if let log = _recentUpdates[ParseRecentUpdates.Log.rawValue] as? String{
            recentLog = Utils.getRecentLogFromJson(log)
        }
        else{
            recentLog = NSMutableArray()
        }
    }
    
    func save() {
        recentUpdates[ParseRecentUpdates.Log.rawValue] = Utils.getJsonFromRecentLog(recentLog)
        do{
            try recentUpdates.save()
        }catch{
        }
    }
}

File: Project/Android/User/app/src/main/java/eduku/org/utils/Models/RecentUpdates.java
package eduku.org.utils.Models;

import com.parse.ParseException;
import com.parse.ParseObject;

import org.json.JSONArray;

import java.util.ArrayList;

import eduku.org.utils.ParseClass.ParseRecentUpdates;
import eduku.org.utils.Utils;

/**
 * Created by mickey on 2/23/16.
 */
public class RecentUpdates {
    public ArrayList<RecentLog> recentLog = null;
    private ParseObject recentUpdates;

    public RecentUpdates(ParseObject _recentUpdates) {
        recentUpdates = _recentUpdates;
        if (_recentUpdates.getString(ParseRecentUpdates.Log) != null){
            recentLog = Utils.getRecentLogFromJson(_recentUpdates.getString(ParseRecentUpdates.Log));
        }
        else{
            recentLog = new ArrayList<>();
        }
    }

    public void save() {
        recentUpdates.put(ParseRecentUpdates.Log, Utils.getJsonFromRecentLog(recentLog));
        try {
            recentUpdates.save();
        } catch (ParseException e) {
            e.printStackTrace();
        }
    }
}
